crates/transaction-pool/src/traits.rs

```diff
@@ -175,6 +175,15 @@ pub trait TransactionPool: Send + Sync + Clone {
         &self,
     ) -> Box<dyn BestTransactions<Item = Arc<ValidPoolTransaction<Self::Transaction>>>>;
 
+    /// Returns an iterator that yields transactions that are ready for block production with the
+    /// given base fee.
+    ///
+    /// Consumer: Block production
+    fn best_transactions_with_base_fee(
+        &self,
+        base_fee: u128,
+    ) -> Box<dyn BestTransactions<Item = Arc<ValidPoolTransaction<Self::Transaction>>>>;
```

Comment:
Starting to worry a little (not due to this PR) about the number of transaction types we have across:
* DB
* TXPool (ValidPoolTransaction, PendingTransaction{Ref} etc.)
* RPC
* ..?

We should document these and during the offsite understand what the overlap looks like/and how you go from one to the other.

Reply:
I can write a primer and put it in the docs or in rpc

Reply:
sg! thx!